### PR TITLE
*: remove some unnecessary dependencies

### LINKED ISSUE #11593: Clean up some unused dependencies
Run `cargo +nightly udeps --all-targets`
```
unused dependencies:
`backup v0.0.1 (/data2/ryan/tikv/components/backup)`
└─── dependencies
     └─── "tokio-stream"
`engine_traits_tests v0.0.1 (/data2/ryan/tikv/components/engine_traits_tests)`
└─── dependencies
     ├─── "engine_test"
     ├─── "engine_traits"
     ├─── "panic_hook"
     ├─── "tempfile"
     └─── "tikv_alloc"
`error_code v0.0.1 (/data2/ryan/tikv/components/error_code)`
└─── dependencies
     └─── "toml"
`external_storage v0.0.1 (/data2/ryan/tikv/components/external_storage)`
└─── dependencies
     ├─── "anyhow"
     └─── "tame-oauth"
`external_storage_export v0.0.1 (/data2/ryan/tikv/components/external_storage/export)`
└─── dependencies
     └─── "anyhow"
`keys v0.1.0 (/data2/ryan/tikv/components/keys)`
└─── dependencies
     └─── "tikv_util"
`raftstore v0.0.1 (/data2/ryan/tikv/components/raftstore)`
└─── dependencies
     └─── "engine_test"
`resource_metering v0.0.1 (/data2/ryan/tikv/components/resource_metering)`
└─── dependencies
     ├─── "prometheus-static-metric"
     └─── "thread-id"
`server v0.0.1 (/data2/ryan/tikv/components/server)`
└─── dependencies
     ├─── "promptly"
     └─── "vlog"
`tikv v5.4.0-alpha (/data2/ryan/tikv)`
└─── dependencies
     ├─── "encryption_export"
     ├─── "engine_test"
     ├─── "engine_traits_tests"
     └─── "hyper-openssl"
`tikv_alloc v0.1.0 (/data2/ryan/tikv/components/tikv_alloc)`
└─── dependencies
     └─── "lazy_static"
`tikv_kv v0.1.0 (/data2/ryan/tikv/components/tikv_kv)`
└─── dependencies
     └─── "vlog"
`tikv_util v0.1.0 (/data2/ryan/tikv/components/tikv_util)`
└─── dependencies
     └─── "fs2"
Note: They might be false-positive.
      For example, `cargo-udeps` cannot detect usage of crates that are only used in doc-tests.
      To ignore some dependencies, write `package.metadata.cargo-udeps.ignore` in Cargo.toml.
```

### PR DESCRIPTION
**Pull request by rleungx**
Signed-off-by: Ryan Leung <rleungx@gmail.com>

### What is changed and how it works?

Close #11593

What's Changed:

Clean up dependencies.

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by sticnarf — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -168,6 +167,7 @@ test_sst_importer = { path = "components/test_sst_importer", default-features =
 test_util = { path = "components/test_util", default-features = false }
 tokio = { version = "1.5", features = ["macros", "rt-multi-thread", "time"] }
 zipf = "6.1.0"
+hyper-openssl = "0.9"
```
Comment (BusyJay) [inline Cargo.toml:170]: Why move the order?

Comment (rleungx) [reply Cargo.toml:170]: It's only used in test.

**Review by BusyJay — APPROVED** [review]

**Pull request merged**